android/app/src/androidTest/java/com/zingo/AddressTest.kt

```diff
@@ -0,0 +1,117 @@
+        val birthday = "1994000"
+        val datadir = ""
+        
+        val saplingSpendFile: InputStream = MainApplication.getAppContext()?.resources?.openRawResource(R.raw.saplingspend)!!
+        var saplingSpend = saplingSpendFile.readBytes()
+        saplingSpendFile.close()
+
+        val middle0 =        0
+        val middle1 =  6000000 // 6_000_000 - 8 pieces
+        val middle2 = 12000000
+        val middle3 = 18000000
+        val middle4 = 24000000
+        val middle5 = 30000000
+        val middle6 = 36000000
+        val middle7 = 42000000
+        val middle8: Int = saplingSpend.size
+        var saplingSpendEncoded = StringBuilder(Base64.encodeToString(saplingSpend, middle0, middle1 - middle0, Base64.NO_WRAP))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle1,
+            middle2 - middle1,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle2,
+            middle3 - middle2,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle3,
+            middle4 - middle3,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle4,
+            middle5 - middle4,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle5,
+            middle6 - middle5,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle6,
+            middle7 - middle6,
+            Base64.NO_WRAP
+        ))
+        saplingSpendEncoded = saplingSpendEncoded.append(Base64.encodeToString(
+            saplingSpend,
+            middle7,
+            middle8 - middle7,
+            Base64.NO_WRAP
+        ))
+
+        saplingSpend = ByteArray(0)
+
+        val saplingOutputFile: InputStream = MainApplication.getAppContext()?.resources?.openRawResource(R.raw.saplingoutput)!!
+        var saplingOutput = saplingOutputFile.readBytes()
+        saplingOutputFile.close()
+
+        val saplingOutputEncoded = StringBuilder(Base64.encodeToString(saplingOutput, Base64.NO_WRAP))
+
+        saplingOutput = ByteArray(0)
+
+        // initlogging()
+
+        val rseed = initfromseed(server, seed, birthday,
+            saplingOutputEncoded.toString(),
+            saplingSpendEncoded.toString(),
+            datadir)
+            // reactContext.applicationContext.filesDir.absolutePath)
+        // Log.w("MAIN", seed)
+    
+        val resp = execute("addresses", "")
+        assertThat(resp).isEqualTo(1)
```

Comment:
I'm not so familiar with Junit... but I believe this assertion always going to be false...

This `resp` is always `string`... and if the value don't start by `Error:` you can safely parse as a JSON abject, and you can read the different values of the `addresses` of this wallet/seed.

Maybe I misunderstanding your goal with this.

Reply:
yes you are right my goal currently is fail the test in a way that gives me an address (which is not equal to 1) ....my issue is... no matter what i try, i can't initialize the lightclient... so it always fails with "Error: lightclient not initialized" (is not equal to 1)... once i return an address i can verify this address is correct and then set the assertion to the address and pass the test

Reply:
i have tried to implement kotlin thread / coroutines but i think i need to go up to the react-native layer, the react-methods are a much nicer way to implement this test, it allows us to write tests for iOS also... and it provides more coverage... this way i can use the same threading etc. to make sure the LC gets initialised. i am new to kotlin etc. so maybe i just missed something! any comments very welcome